AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -658,8 +705,40 @@ - (void)dataControllerUnlockDataSource
   }
 }
 
-#pragma mark -
-#pragma mark ASRangeControllerDelegate.
+#pragma mark - ASCollectionViewDataControllerSource Supplementary view support
+
+- (ASDisplayNode *)dataController:(ASCollectionDataController *)dataController supplementaryNodeOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
+{
+  ASDisplayNode *node = [_asyncDataSource collectionView:self nodeForSupplementaryElementOfKind:kind atIndexPath:indexPath];
+  ASDisplayNodeAssert(node != nil, @"A node must be returned for a supplementary node");
+  ASDisplayNodeAssert(!node.nodeLoaded, @"The supplementary node must not be loaded");
```

Comment:
Sounds good, leave it in!  @Adlai-Holler's diff just landed (which will create a small conflict with this one, should be easy to revolve) and should support loaded nodes now.

There are some great ways to improve the behavior of loaded nodes, to allow async measure and such.  Hope to get to those in the coming months.